### ISSUE TITLE
TypeError: Cannot read property 'isExternal' of null

### ISSUE THREAD

**Issue by domoritz** [rollup/rollup#2275]
Clone https://github.com/vega/vega-lite/tree/dom/rollup-bug, then run

Rollup: 0.60.7. No problems in 0.59. 

```
yarn
yarn build:only
```

Error

```
yarn run v1.7.0
$ tsc && cp package.json build/src/ && rollup -c
build/src/index.js → build/vega-lite.js...
[!] TypeError: Cannot read property 'isExternal' of null
TypeError: Cannot read property 'isExternal' of null
    at Module.traceVariable (/Users/domoritz/Developer/UW/vega-lite/node_modules/rollup/dist/rollup.js:13092:30)
    at ModuleScope.findVariable (/Users/domoritz/Developer/UW/vega-lite/node_modules/rollup/dist/rollup.js:12366:29)
    at FunctionScope.Scope.findVariable (/Users/domoritz/Developer/UW/vega-lite/node_modules/rollup/dist/rollup.js:3925:68)
    at Scope.findVariable (/Users/domoritz/Developer/UW/vega-lite/node_modules/rollup/dist/rollup.js:3925:68)
    at FunctionScope.Scope.findVariable (/Users/domoritz/Developer/UW/vega-lite/node_modules/rollup/dist/rollup.js:3925:68)
    at Scope.findVariable (/Users/domoritz/Developer/UW/vega-lite/node_modules/rollup/dist/rollup.js:3925:68)
    at BlockScope.Scope.findVariable (/Users/domoritz/Developer/UW/vega-lite/node_modules/rollup/dist/rollup.js:3925:68)
    at Identifier$$1.bind (/Users/domoritz/Developer/UW/vega-lite/node_modules/rollup/dist/rollup.js:9566:40)
    at Identifier$$1.reassignPath (/Users/domoritz/Developer/UW/vega-lite/node_modules/rollup/dist/rollup.js:9628:18)
    at LocalVariable.reassignPath (/Users/domoritz/Developer/UW/vega-lite/node_modules/rollup/dist/rollup.js:2628:27)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by domoritz** [comment]
Thank you for looking at the issues and the thorough analysis @guybedford.

**Comment by lukastaegert** [comment]
> Not sure what the best fix would be yet

There is exactly one situations where `reassignPath` can be called during initialisation (and not during bind phase) which might be what is happening here. This is when a variable has more than one declaration. I.e. in this situation:

```js
var x = {a: 1};
var x = {b: 2};
```

the second line counts as an assignment of a new value to `x`.  However as the second line does not know it actually is just an assignment, the necessary reassignment currently happens during the initialisation phase which apparently can trip up rollup in specific situations.

I hope I can look a little deeper into how this could be solved tomorrow.

**Comment by guybedford** [comment]
@lukastaegert I was getting the issue in this file - https://github.com/vega/vega-lite/blob/master/src/compile/layout/header.ts, for either the `HEADER_CHANNELS` or `HEADER_TYPES` variable (I can't recall which offhand now). Could it be the `for of` statement perhaps?

**Comment by lukastaegert** [comment]
Thanks for digging this up @guybedford!

**Comment by lukastaegert** [comment]
My bet is on getHeaderGroup being declared twice.

**Comment by lukastaegert** [comment]
Sure it is `HEADER_TYPES `?

**Comment by guybedford** [comment]
It is definitely `HEADER_CHANNELS` so the `for of` statement must be the cause somehow.

**Comment by guybedford** [comment]
Seems like it is imported at https://github.com/vega/vega-lite/blob/30be178cf9f98c27deb27433980ff36bc4d07084/src/compile/model.ts#L22.

**Comment by lukastaegert** [comment]
But this is inside a class. Might be necessary to look at the transpiled TypeScript code I fear.

**Comment by guybedford** [comment]
The issue seems to be happening for the importer source, the the exporting source.

Transpiled code - https://gist.github.com/guybedford/c936cbd60f38c716503c7191873e474d.

**Comment by guybedford** [comment]
Assignment at https://gist.github.com/guybedford/c936cbd60f38c716503c7191873e474d#file-header-js-L209.

**Comment by lukastaegert** [comment]
Ok, you were spot on. In fact, it was the double declaration of `channel` (i.e. the loop variable) that was only counted as a double declaration because in the transpiled source, `const` was replaced with hoisted `var` statements (would be interesting to see if playing around with this transpilation a little more could actually trigger a TypeScript bug).

Fix is upcoming, I discovered a few other related issues while going over this...

**Comment by 23d1** [comment]
I seem to be running into this issue now trying to compile when importing 'smooth-scrolling', a repo that works great in webpack as well as parcel. Any ideas how I might resolve?

**Comment by lukastaegert** [comment]
Hi @23d1,

As for the error when running `rollup -c -w`, the "Unexpected token" in "init.json" looks like you need `rollup-plugin-json`.
As for the original error: I fear this will not be easy to debug without a reproduction, e.g. a repository with this issue. Could you check that you are using the latest version of Rollup that contains the fix for the current issue, ideally rollup@0.66.0 ?
